### ISSUE TITLE
Mangling properties with decorators

### ISSUE THREAD

**Issue by bo4arovpavel1989** [terser/terser#1075]
Hello! I would like to ask a piece of advice. 

We use terser to minify private properties of classes in typescript. Minifying of private properties is executed by marking private properties with special `_private` prefix (using package [ts-transformer-minify-privates](https://www.npmjs.com/package/ts-transformer-minify-privates)) and passing appropriate regex to terser's mangle properties regex option.

 It works fine, except case when using decorators for private class methods, since propertyKey argument of decorator is passed as not-mangled method name. 

So for now we have to avoid mangling decorated methods (fot instance, by declaring them as protected, thus ts-transformer-minify-privates doesn't prefix them), but maybe there might be a better way and mangling would work with decorators as well?

**Comment by jridgewell** [comment]
I'm assuming you're using the output from TypeScript's [decorators transform](https://www.typescriptlang.org/play?emitDecoratorMetadata=false&jsx=1&ssl=5&ssc=1&pln=5&pc=2#code/PTAEHcCcEsBcFNQHtKgA4EMDODQGMkATRAC3kngChK8AbbLUAMSSVAG9LRQABAM1boYANwy4+0eLUKgAvKACMAbkoBfIA)?

```js
// input
class Foo {
  @foo private field = 1;
}

// output
class Foo {
    constructor() {
        this.field = 1;
    }
}
__decorate([
    foo
], Foo.prototype, "field", void 0);
```

The issue is that TypeScripts decorators are highly dynamic and **non-standard**. There's no way good way to relate the `"field"` to `this.field`.

There's request to enable mangling string literals at https://github.com/terser/terser/issues/990. It's we could synchronize the string/identifier mangling so the same inputs get the same results.

**Comment by fabiosantoscode** [comment]
> There's request to enable mangling string literals at #990. It's we could synchronize the string/identifier mangling so the same inputs get the same results.

Side note: should the variable mangling be synchronized as well? If we're going to sync those two, why not sync a third one. There's a marginal improvement in compression to be had, and potentially some readability benefits.

**Comment by bo4arovpavel1989** [comment]
> There's request to enable mangling string literals at #990. It's we could synchronize the string/identifier mangling so the same inputs get the same results.

Yes, that's the exact output. Seems, it could be a real solution, i'll keep an eye on changes, thanks a lot)
